### PR TITLE
Updates on Driver and passenger's dashboards

### PR DESCRIPTION
**Pull request by BaharaJr**
I updated the dashboards to capture more cases and functionalities that hadn't been caught before.

### CODE REVIEW

File: frontend/src/features/Driver/rideoffer.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import './driver.css';
+import { Link } from 'react-router-dom';
+
+class Rideoffers extends Component {
+    render() {
```
Comment (BenMaruchu) [inline frontend/src/features/Driver/rideoffer.js]: This component doesn't have lifecycle methods you can change it to functional/ stateless component

File: frontend/src/features/Driver/riderequest.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import './driver.css';
+import { Link } from 'react-router-dom';
+
+class RideRequests extends Component {
+    render() {
```
Comment (BenMaruchu) [inline frontend/src/features/Driver/riderequest.js]: This also change it to functional component

File: frontend/src/features/passenger/profile.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import './profile.css'
+import { Link } from 'react-router-dom';
+
+class Profile extends Component {
```
Comment (BenMaruchu) [inline frontend/src/features/passenger/profile.js]: Change this component also to functional component

File: frontend/src/features/passenger/rides.js [hunk 4]
```diff
@@ -1,32 +1,31 @@
...
-import './profile.css'
+import React, { Component } from 'react';
+import './profile.css';
 import { Link } from 'react-router-dom';
 
-class Profile extends Component{
-
-    render(){
-        return(
+class Rides extends Component {
```
Comment (BenMaruchu) [inline frontend/src/features/passenger/rides.js]: change this one also to functional component

File: frontend/src/features/passenger/taken.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
+import React, {Component} from 'react';
+import './profile.css'
+import { Link } from 'react-router-dom';
+
+class Taken extends Component{
```
Comment (BenMaruchu) [inline frontend/src/features/passenger/taken.js]: Change this one also to functional component

**Review by BenMaruchu — CHANGES_REQUESTED** [review]
Hello @BaharaJr,

Nice work,

Here are couple of things you can improve

1. Please configure your editor to format your codes
2. Change all class components which don't have lifecycle methods to functional compoents


Otherwise nice work, I will be expecting changes from you

**Comment by BaharaJr** [comment]
> Hello @BaharaJr,
> 
> Nice work,
> 
> Here are a couple of things you can improve
> 
> 1. Please configure your editor to format your codes
> 2. Change all class components which don't have lifecycle methods to functional components
> 
> Otherwise nice work, I will be expecting changes from you

Thank you, Ben. I will work on them and send them before the end of the day

**Comment by BenMaruchu** [comment]
ok @BaharaJr

**Pull request closed without merging**